e2e/test/scenarios/joins/reproductions/23293-drill-through-implicit-join-add-remove-column.cy.spec.js

```diff
@@ -0,0 +1,96 @@
+import {
+  restore,
+  popover,
+  openOrdersTable,
+  visitDashboard,
+} from "e2e/support/helpers";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+
+const { ORDERS, PRODUCTS } = SAMPLE_DATABASE;
+
+describe("issue 23293", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should retain the filter when drilling through the dashboard card with implicitly added column (metabase#23293)", () => {
+    openOrdersTable();
+
+    cy.findByTestId("viz-settings-button").click();
+    modifyColumn("Product ID", "remove");
+    modifyColumn("Category", "add");
+    cy.wait("@dataset");
+
+    saveQuestion().then(
+      ({
+        response: {
+          body: { id },
+        },
+      }) => {
+        const questionDetails = {
+          query: {
+            "source-table": `card__${id}`,
+            aggregation: [["count"]],
+            breakout: [
+              [
+                "field",
+                PRODUCTS.CATEGORY,
+                {
+                  "source-field": ORDERS.PRODUCT_ID,
+                },
+              ],
+            ],
+          },
+          display: "bar",
+        };
+
+        cy.createQuestionAndDashboard({ questionDetails }).then(
+          ({ body: { dashboard_id } }) => {
+            visitDashboard(dashboard_id);
+          },
+        );
+
+        cy.get(".bar").first().realClick();
+        popover()
+          .findByText(/^See these/)
+          .click();
+
+        cy.findByTestId("qb-filters-panel").should(
+          "contain",
+          "Product → Category is Doohickey",
+        );
+        cy.findAllByTestId("header-cell")
+          .last()
+          .should("have.text", "Product → Category");
+
+        cy.findAllByRole("grid")
+          .last()
+          .as("tableResults")
```

Comment:
are you using this alias at all? If not can we remove it?

Reply:
We are not using this alias anywhere, but you'll be glad it exists when and if you come back to this test after six months and start wondering what is the "last grid"

Reply:
That makes sense 👍